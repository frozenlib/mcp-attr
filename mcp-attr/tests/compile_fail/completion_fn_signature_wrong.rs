use mcp_attr::{server::{McpServer, mcp_server}, Result};

struct TestServer;

#[mcp_server]
impl McpServer for TestServer {
    #[prompt]
    async fn test_prompt(&self, #[complete(.complete_wrong)] msg: String) -> Result<String> {
        Ok(msg)
    }
    
    #[complete_fn]
    async fn complete_wrong(&self) -> Result<Vec<String>> {
        //~^ ERROR: completion function must have `value: &str` parameter
        Ok(vec!["test".to_string()])
    }
}

fn main() {}