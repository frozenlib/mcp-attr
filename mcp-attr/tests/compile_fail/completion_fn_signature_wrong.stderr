error: completion function `complete_wrong` must have a `value: &str` parameter
  --> tests/compile_fail/completion_fn_signature_wrong.rs:13:14
   |
13 |     async fn complete_wrong(&self) -> Result<Vec<String>> {
   |              ^^^^^^^^^^^^^^

error[E0061]: this method takes 0 arguments but 2 arguments were supplied
  --> tests/compile_fail/completion_fn_signature_wrong.rs:8:45
   |
8  |     async fn test_prompt(&self, #[complete(.complete_wrong)] msg: String) -> Result<String> {
   |                                             ^^^^^^^^^^^^^^
   |                                             |
   |                                             unexpected argument #1 of type `&CompleteRequestParams`
   |                                             unexpected argument #2 of type `&mut mcp_attr::server::RequestContext`
   |
note: method defined here
  --> tests/compile_fail/completion_fn_signature_wrong.rs:13:14
   |
13 |     async fn complete_wrong(&self) -> Result<Vec<String>> {
   |              ^^^^^^^^^^^^^^
help: remove the extra arguments
   |
8  -     async fn test_prompt(&self, #[complete(.complete_wrong)] msg: String) -> Result<String> {
8  +     async fn test_prompt(&self, #[complete(.g)] msg: String) -> Result<String> {
   |

error[E0308]: mismatched types
 --> tests/compile_fail/completion_fn_signature_wrong.rs:8:45
  |
8 |     async fn test_prompt(&self, #[complete(.complete_wrong)] msg: String) -> Result<String> {
  |                                             ^^^^^^^^^^^^^^ expected `Result<CompleteResult, Error>`, found `Result<Vec<String>, Error>`
  |
  = note: expected enum `std::result::Result<mcp_attr::schema::CompleteResult, mcp_attr::Error>`
             found enum `std::result::Result<Vec<std::string::String>, mcp_attr::Error>`
